pkg/canary/comparator/comparator.go

```diff
@@ -186,12 +244,116 @@ func (c *Comparator) run() {
 			c.entrySent(e)
 		case <-t.C:
 			c.pruneEntries()
+		case <-sc.C:
```

Comment:
I wonder if you shouldn't create a goroutine per check. If a spot check or a metric query is taking time, then you might slow other tick ?

Reply:
yeah that was existing behavior but I took this approach for the new checks for the reason you just specified, I will make that change for `pruneEntries` as well